### DISCUSSION TITLE
invalid DNS upstream issue

### DISCUSSION THREAD

**Discussion by T-Mwangi** [AdguardTeam/AdGuardHome#7246] [Q&A]
Hello all,

I seem to be having an issue using any DNS server from my list of DNS servers in "Setup Guide"

Context:

1. I'm following the instal/set up guide found [here](https://pimylifeup.com/raspberry-pi-adguard-home/).
2. AdGuard Home is running correctly.
3. I can connect to the DNS server using a MacBook or windows laptop (via their official AdGuard apps) and see DNS queries on AdGuard Home, so the DNS is working.
4. All devices are on the same Wi-Fi network.

However, when trying to configure the DNS on my iPhone or iPad, AdGuard doesn't seem to register these devices and throws an "invalid upstream" error. I'm using the official AdGuard apps to configure the DNS on these devices (DNS protections -> DNS Server), but I encounter the error when using any of the listed DNS servers. For example, I'm using my own server, but the issue occurs with all servers listed. reinstalling the apps on the iPad or iPhone doesn't resolve the issue, and I recently reinstalled AdGuardHome, and this hasn't resolved the issue.

This problem persists even after waiting several hours for the devices to potentially recognise the necessary port. If it does work temporarily, it stops working after disconnecting from the DNS server (e.g., when I leave the network) and then reconnecting. I am aware that I can set it up via the router, but I'd like to ensure it works correctly before committing to configuring it on the router.

Does anyone have any suggestions or insights on this issue?

### Ipad
<img width="1194" alt="image" src="https://github.com/user-attachments/assets/6a19242a-b5c4-403f-bd97-2a6bbae3b580">

### iPhone
![image](https://github.com/user-attachments/assets/e0c39c07-3d03-49c5-a452-95d043728f65)

### AdGuard server
<img width="1194" alt="image" src="https://github.com/user-attachments/assets/22745c1e-3944-41c3-b589-bb2ff0f06400">

**Comment by Dynamic5912** [comment]
The AdGuard App has nothing to do with AdGuard Home.

**Reply by T-Mwangi** [reply]
Sure, but even if I delete the AdGuard app on the iPhone and iPad, I still have issues with using the AdGuard Home DNS. 

Without the app, even if I explicitly add every DNS IP to my WiFi DNS server list, AdGuard Home will still not recognise the devices. This basically becomes a silent rejection. I was using the app, because for whatever reason, the desktop apps seem to have no issue with recognising the given DNS Servers. 

If you have a suggestion to why the DNS servers will be silently rejected even when adding the DNS IP's to my WiFi's DNS list, I'm more that happy to try out

**Reply by Dynamic5912** [reply]
You just need to add the AGH IP as the LAN DNS on your router.

Then all devices will be routed to there for DNS.

**Reply by T-Mwangi** [reply]
I am aware that setting it up via the router is an option, and I do plan on doing that at some point, but setting it up per devices, according to AGH, is also an option.

I don't want to implement AGH to my router, without first ensuring that it works on other devices, especially if setting it up per devices is explicitly an option to do. I also want to understand why this isn't working, so that if there are potential issues with AGH, I'm not potentially affecting other users on the network.

**Reply by Dynamic5912** [reply]
Where have you setup AGH?

Is it reachable (ping/tracert etc) from the LAN your devices are connected to?

**Reply by T-Mwangi** [reply]
My AGH is running off a raspberry pi 5, which is connected via Ethernet to my router. I'm connected to the router on my phone and iPad via WiFi. 

I can connect to the AGH dashboard page via my phone's browser, so I know it's not a connection issue, which is why I'm confused on why it's not connecting.

I even made sure when selecting the DNS port that it wasn't a port that was considered "used" (via the setup screen), so as far as I'm aware it's using an available port. Considering I can connect to the DNS via a computer, I know it's not an issue with the port.

**Reply by Dynamic5912** [reply]
So does it work if you set it to your routers LAN IP DNS ?

**Reply by T-Mwangi** [reply]
I haven't done that, because:

1. I'm not the only person using the router, and while I have plans on setting the DNS router wide, if there is an issue with AGH, or it's filtering rules, I prefer to deal with it on my own devices, rather than dealing with everyone else's devices. 
2. AGH through its own page, does give instructions on how to set it up per device. If this was a case of me trying to force my device to do something it couldn't do, I would just set it up via the router. The fact that a process that should work, which does work on other devices, doesn't work via my phone is either a bug in the software, or bug in the installation. I'm more interested in determining which is the case. 

I'm sure running it via the router is a preferred method, but I'm trying to figure out this issue, rather than bypassing it by setting up AGH router wide.
